app/lib/katello/resources/candlepin.rb

```diff
@@ -197,7 +197,7 @@ def refresh_entitlements(uuid)
 
           def consume_entitlement(uuid, pool, quantity = nil)
             uri = join_path(path(uuid), 'entitlements') + "?pool=#{pool}"
-            uri += "&quantity=#{quantity}" if quantity && quantity > 0
+            uri += "&quantity=#{quantity}" if quantity && quantity.to_i > 0
```

Comment:
You could drop the initial check and shorten to `if quantity.to_i > 0` since nil.to_i = 0